### PR TITLE
Remove base price from dashboard

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because it
- migrates product base price to gross price from priceRangeUndiscounted in the product list and category product list
- removes base price usage from product create and product update page
- removes base price usage from product variant creator

**PR intended to be tested with API branch:** 

### Screenshots

Zrzut ekranu 2020-06-16 o 18 54 29
Zrzut ekranu 2020-06-16 o 18 55 20

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: src/products/queries.ts [hunk 1]
```diff
@@ -115,6 +112,12 @@ const productVariantAttributesFragment = gql`
         }
       }
     }
+    variants {
+      price {
```
Comment (dominik-zeglen) [inline src/products/queries.ts:116]:
```suggestion
    variants {
      id
      price {
```

File: src/products/queries.ts [hunk 2]
```diff
@@ -115,6 +112,12 @@ const productVariantAttributesFragment = gql`
...
+        amount
+        currency
```
Comment (dominik-zeglen) [inline src/products/queries.ts:118]:
```suggestion
        ...Money
```

File: src/products/queries.ts [hunk 3]
```diff
@@ -337,6 +337,12 @@ const productListQuery = gql`
               name
             }
           }
+          variants {
+            price {
```
Comment (dominik-zeglen) [inline src/products/queries.ts:341]:
```suggestion
          variants {
            id
            price {
```

File: src/products/components/ProductListPage/utils.ts [hunk 4]
```diff
@@ -11,3 +13,21 @@ export function isAttributeColumnValue(value: string) {
...
+    const price = productVariants[i].price;
+    max = price.amount > max ? price.amount : max;
+    min = price.amount < min ? price.amount : min;
+  }
+
+  return {
+    max,
+    min
+  };
+}
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/products/components/ProductListPage/utils.ts:33]: Are you sure? It replaces one `for loop` with three `max, min and map loops`, it's less efficient - https://codeburst.io/javascript-finding-minimum-and-maximum-values-in-an-array-of-objects-329c5c7e22a2

File: src/products/components/ProductList/ProductList.tsx [hunk 5]
```diff
@@ -61,9 +62,17 @@ const useStyles = makeStyles(
       }
     },
     colNameFixed: {},
+    colNameGrid: {
+      display: "grid",
+      gridColumn: 1
+    },
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/products/components/ProductList/ProductList.tsx:72]: hmm, in order to have two rows. I've changed it to display block

File: src/products/components/ProductListPage/utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply src/products/components/ProductListPage/utils.ts]: @orzechdev 
lets say, there are 25 products on page.
Each with 10 variants. 
Thats really, really, small processing power needing for calculation. Perf links from article dont work, so cant estimate amount of microseconds difference.

File: locale/defaultMessages.json [hunk 6]
```diff
@@ -3567,6 +3567,9 @@
     "context": "product price",
     "string": "Price"
   },
+  "src_dot_products_dot_components_dot_ProductList_dot_150865454": {
+    "string": "Simple"
```
Comment (dominik-zeglen) [inline locale/defaultMessages.json:3571]: Adjectives should always have context provided - eg in polish, would you translate it as "prosty", "proste" or "prosta"?

File: locale/ar.json [hunk 7]
```diff
@@ -560,7 +560,7 @@
   },
   "src_dot_back": {
     "context": "button",
-    "string": "Back"
+    "string": "عودة"
```
Comment (krzysztofwolski) [inline locale/ar.json:563]: Why there are translation changes in this PR?

File: locale/defaultMessages.json [hunk 8]
```diff
@@ -3578,6 +3581,9 @@
     "context": "product status",
     "string": "Not published"
   },
+  "src_dot_products_dot_components_dot_ProductList_dot_2754779425": {
+    "string": "Configurable"
```
Comment (dominik-zeglen) [inline locale/defaultMessages.json:3585]: Same here

File: src/products/components/ProductList/ProductList.tsx [hunk 9]
```diff
@@ -296,7 +344,22 @@ export const ProductList: React.FC<ProductListProps> = props => {
                     thumbnail={maybe(() => product.thumbnail.url)}
                     data-tc="name"
                   >
-                    {maybe<React.ReactNode>(() => product.name, <Skeleton />)}
+                    {product?.productType ? (
+                      <div className={classes.colNameWrapper}>
+                        <span>{product.name}</span>
+                        {product && product.productType && (
```
Comment (dominik-zeglen) [inline src/products/components/ProductList/ProductList.tsx:350]:
```suggestion
                        {product?.productType && (
```

File: src/products/components/ProductList/ProductList.tsx [hunk 10]
```diff
@@ -130,6 +138,46 @@ export const ProductList: React.FC<ProductListProps> = props => {
...
   const numberOfColumns = 2 + settings.columns.length;
 
+  const getProductPrice = (
+    variants: ProductList_products_edges_node_variants[]
+  ) => {
+    if (!variants.length) {
+      return null;
+    }
+
+    const { max, min } = getProductPriceRange(variants);
```
Comment (krzysztofwolski) [inline src/products/components/ProductList/ProductList.tsx:148]: But product will still have pricerange field at product AFAIK. Why not use it?

Comment (krzysztofwolski) [reply src/products/components/ProductList/ProductList.tsx:148]: image

Comment (orzechdev) [reply src/products/components/ProductList/ProductList.tsx:148]: I didn't know about it. I'll change it

File: locale/ar.json (hunk 7 above)
Comment (orzechdev) [reply locale/ar.json:563]: because of the mistaken transifex push

File: locale/defaultMessages.json (hunk 6 above)
Comment (orzechdev) [reply locale/defaultMessages.json:3571]: sure, I didn't know about it

File: src/products/components/ProductList/ProductList.tsx (hunk 10 above)
Comment (orzechdev) [reply src/products/components/ProductList/ProductList.tsx]: fixed, changed to priceRangeUndiscounted and gross price is displayed

File: src/products/components/ProductListPage/utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/products/components/ProductListPage/utils.ts]: fixed https://github.com/mirumee/saleor-dashboard/pull/555#discussion_r442295862

File: src/categories/queries.ts [hunk 11]
```diff
@@ -8,6 +8,13 @@ import {
 } from "./types/CategoryDetails";
 import { RootCategories } from "./types/RootCategories";
 
+export const fragmentMoney = gql`
```
Comment (dominik-zeglen) [inline src/categories/queries.ts:11]: `src/products/queries.ts:48` here's the same fragment

Comment (orzechdev) [reply src/categories/queries.ts]: fixed

**Pull request merged**